### PR TITLE
[Release] Add multi-node, multi-GPU SGD release test

### PR DESCRIPTION
**Pull request by amogkam**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/nightly_tests.yaml [hunk 1]
```diff
@@ -72,6 +72,21 @@
...
+    slack: "@tune-team"
+
+  cluster:
+    app-config: sgd_gpu/sgd_gpu_app_config.yaml
+    compute_template: sgd_gpu/sgd_gpu_compute.yaml
+
+  run:
+    timeout: 3000
+    prepare: python wait_cluster.py 2 600
+    script: python sgd_gpu/sgd_gpu_test.py --num-workers=2 --use-gpu --address=auto
```
Comment (amogkam) [inline release/nightly_tests/nightly_tests.yaml:88]:
```suggestion
    script: python sgd_gpu/sgd_gpu_test.py
```

File: release/nightly_tests/sgd_gpu/sgd_gpu_app_config.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+python:
+  pip_packages: []
+  conda_packages: []
+
+post_build_cmds:
+  - pip uninstall -y ray
+  - pip3 install -U {{ env["RAY_WHEELS"] | default("ray") }}
+  - pip3 install -U ray[tune]
+  - echo {{env["DATESTAMP"]}}
```
Comment (krfricke) [inline release/nightly_tests/sgd_gpu/sgd_gpu_app_config.yaml:13]: We  usually don't need this line (DATESTAMP) when we install wheels.

File: release/nightly_tests/nightly_tests.yaml [hunk 3]
```diff
@@ -72,6 +72,21 @@
...
+
+  cluster:
+    app-config: sgd_gpu/sgd_gpu_app_config.yaml
+    compute_template: sgd_gpu/sgd_gpu_compute.yaml
+
+  run:
+    timeout: 3000
+    prepare: python wait_cluster.py 2 600
+    script: python sgd_gpu/sgd_gpu_test.py
+
```
Comment (krfricke) [inline release/nightly_tests/nightly_tests.yaml:89]: Let's move this into a different file. The `nightly_tests` is misleading (these are the "core nightly tests") and we should probably rename them (I can do that later). Let's move this into a separate sgd tests file

File: release/nightly_tests/sgd_gpu/sgd_gpu_test.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    try:
+        train_cifar(num_workers=2, use_gpu=True, num_epochs=5, fp16=True,
+                    test_mode=False)
+    except Exception as e:
+        print(f"The test failed with {e}")
+        success = False
+
+    delta = time.time() - start_time
+    with open(os.environ["TEST_OUTPUT_JSON"], "w") as f:
+        f.write(json.dumps({"shuffle_time": delta, "success": success}))
```
Comment (krfricke) [inline release/nightly_tests/sgd_gpu/sgd_gpu_test.py:21]: Is `shuffle_time` a good name? Shouldn't this be something like `train_time` or so?

**Review by krfricke — CHANGES_REQUESTED** [review]
Generally looks good, but left a couple of minor comments

File: release/nightly_tests/sgd_gpu/sgd_gpu_test.py (hunk 4 above)
Comment (amogkam) [reply release/nightly_tests/sgd_gpu/sgd_gpu_test.py:21]: Ah yes good catch! I copied this over from the shuffle test and forgot to change it.

File: release/nightly_tests/nightly_tests.yaml (hunk 3 above)
Comment (amogkam) [reply release/nightly_tests/nightly_tests.yaml:89]: I created a new directory called "nightly_gpu". This can also be place where we can add Horovod on Ray or Ray Lightning tests.

**Comment by amogkam** [comment]
Thanks for the review! Comments are addressed

**Review by krfricke — APPROVED** [review]
LGTM
